### PR TITLE
Glob pattern matching within a GCS backend

### PR DESCRIPTION
**Pull request by bschaatsbergen**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | yes/**no**
| 🚀 New feature?   | **yes**/no
| ⚠ Deprecations?   | yes/**no**
| ❌ BC Break       | yes/**no**
| 🔗 Related issues | #1330
| ❓ Documentation  | **yes** 

## Description

Adds the ability to make use of glob pattern matching within the GS Backend.

### CODE REVIEW

**Comment by bschaatsbergen** [comment]
Glob pattern matching seems to work, need to figure out why setting a prefix without a `*` doesn't work yet.

**Comment by bschaatsbergen** [comment]
Pending: tests

**Comment by bschaatsbergen** [comment]
I've removed the tests and raised a new issue to improve the mocking of Google Cloud clients and APIs.

File: pkg/iac/terraform/state/enumerator/glob.go [hunk 1]
```diff
@@ -15,6 +15,11 @@ func GlobS3(path string) (prefix string, pattern string) {
 	return
 }
 
+func GlobGS(path string) (prefix string, pattern string) {
```
Comment (moadibfr) [inline pkg/iac/terraform/state/enumerator/glob.go:18]: it looks like this and a good part of the enumeration method is duplicated from the s3 code. Do you think we could find a way to share the mathcing code ?

Comment (bschaatsbergen) [reply pkg/iac/terraform/state/enumerator/glob.go:18]: I was thinking the same to be honest, I'll take a look at it this evening and see if I can offload some code to common functions 👍🏼

Comment (bschaatsbergen) [reply pkg/iac/terraform/state/enumerator/glob.go]: I've moved some code around, to a `common.go`, removed some wrapper functions that didn't add a lot of value.

Comment (bschaatsbergen) [reply pkg/iac/terraform/state/enumerator/glob.go]: Changed nothing in the functionality, just renamed functions and removed duplicate code.

**Review by moadibfr — APPROVED** [review]
LGTM thanks a lot for your contribution !
Could you also please do some documentation on this on our documentation repo ?

**Comment by bschaatsbergen** [comment]
Definitely, will update the PR shortly

**Comment by bschaatsbergen** [comment]
> LGTM thanks a lot for your contribution ! Could you also please do some documentation on this on our documentation repo ?

See https://github.com/snyk/driftctl-docs/pull/263

**Comment by moadibfr** [comment]
@all-contributors please add @bschaatsbergen for code and documentation

**Pull request merged**